src/Protocol/Diaspora/Entity/DiasporaContact.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Friendica\Protocol\Diaspora\Entity;
+
+use Psr\Http\Message\UriInterface;
+
+/**
+ * @property-read $url
+ * @property-read $uriId
+ * @property-read $guid
+ * @property-read $addr
+ * @property-read $alias
+ * @property-read $nick
+ * @property-read $name
+ * @property-read $givenName
+ * @property-read $familyName
+ * @property-read $photo
+ * @property-read $photoMedium
+ * @property-read $photoSmall
+ * @property-read $batch
+ * @property-read $notify
+ * @property-read $poll
+ * @property-read $subscribe
+ * @property-read $searchable
+ * @property-read $pubKey
+ * @property-read $baseurl
+ * @property-read $gsid
+ * @property-read $updated
+ */
+class DiasporaContact extends \Friendica\BaseEntity
+{
+	/** @var UriInterface */
+	protected $url;
+	/** @var int */
+	protected $uriId;
+	/** @var string */
+	protected $guid;
+	/** @var string */
+	protected $addr;
+	/** @var UriInterface */
+	protected $alias;
+	/** @var string */
+	protected $nick;
+	/** @var string */
+	protected $name;
+	/** @var string */
+	protected $givenName;
+	/** @var string */
+	protected $familyName;
+	/** @var UriInterface */
+	protected $photo;
+	/** @var UriInterface */
+	protected $photoMedium;
+	/** @var UriInterface */
+	protected $photoSmall;
+	/** @var UriInterface */
+	protected $batch;
+	/** @var UriInterface */
+	protected $notify;
+	/** @var UriInterface */
+	protected $poll;
+	/** @var UriInterface */
+	protected $subscribe;
+	/** @var bool */
+	protected $searchable;
+	/** @var string */
+	protected $pubKey;
+	/** @var UriInterface */
+	protected $baseurl;
+	/** @var int */
+	protected $gsid;
+	/** @var \DateTime */
+	protected $updated;
+
+	public function __construct(
+		UriInterface $url, string $uriId = null, string $guid = null, string $addr = null, UriInterface $alias = null,
```

Comment:
The uriId should be an integer value.

Reply:
Good catch again, the goal is to make the parameter type as strict as possible, since we are depending on PSR-7 for the HTTP client we can now use the PSR-7 `UriInterface` interface for all the parameters meant to hold a `Uri` thanks to the `Guzzle` implementation. These objects can be easily converted to strings if needed and can remove the need for `parse_url` and especially `Network::unparseUrl`.